### ISSUE TITLE
Insufficient escape in first name

### ISSUE THREAD

**Issue by nhahtdh** [callumacrae/tryregex#18]
From https://github.com/callumacrae/tryregex/blob/master/app/assets/js/globalFuncs.js#L72

```
data.firstEscaped = firstName.replace(/([$()*+.?\[^|\]])/g, '\\$1');
```

It is missing escape for `{`, `}`, `\` and `/` (the last one for RegExp literal only). The code could be changed to:

```
data.firstEscaped = firstName.replace(/([$()*+.?\[^|\]{}\/\\])/g, '\\$1');
```

Or without using capturing group:

```
data.firstEscaped = firstName.replace(/[$()*+.?\[^|\]{}\/\\]/g, '\\$&');
```

**Comment by callumacrae** [comment]
Sure. Does this fix any bugs? Are there any cases where the name is escaped improperly?

**Comment by nhahtdh** [comment]
Sure, it's very unlikely anyone has this name, but since you escaped all other characters, why not go for a full coverage?

```
setName("lol!@#$%^&*()_+-={}[]|\\:\";'<>?,./")
```

**Comment by callumacrae** [comment]
fair enough! would you be able to send a PR? Otherwise, I'll do it later.

Thanks!
